custom_components/owlintuition/sensor.py

```diff
@@ -103,7 +103,7 @@
     SENSOR_ELECTRICITY_RADIO: ['Electricity Radio', SIGNAL_STRENGTH_DECIBELS_MILLIWATT, 'mdi:signal', OWLCLASS_ELECTRICITY, SensorDeviceClass.SIGNAL_STRENGTH, SensorStateClass.MEASUREMENT],
     SENSOR_ELECTRICITY_POWER: ['Electricity Power', UnitOfPower.WATT, 'mdi:flash', OWLCLASS_ELECTRICITY, SensorDeviceClass.POWER, SensorStateClass.MEASUREMENT],
     SENSOR_ELECTRICITY_ENERGY_TODAY: ['Electricity Today', UnitOfEnergy.KILO_WATT_HOUR, 'mdi:flash', OWLCLASS_ELECTRICITY, SensorDeviceClass.ENERGY, SensorStateClass.TOTAL_INCREASING],
-    SENSOR_ELECTRICITY_COST_TODAY: ['Cost Today', None, 'mdi:coin', OWLCLASS_ELECTRICITY, SensorDeviceClass.MONETARY, None],
+    SENSOR_ELECTRICITY_COST_TODAY: ['Cost Today', None, 'mdi:coin', OWLCLASS_ELECTRICITY, SensorDeviceClass.MONETARY, SensorStateClass.TOTAL],
```

Comment:
Should it be `SensorStateClass.TOTAL_INCREASING` rather than `SensorStateClass.TOTAL` ?

Reply:
Interesting…. Was my view too and tested as Total_Increasing and throws a different HA error. See this thread.  https://github.com/home-assistant/core/issues/115692
I then changed to total and there is no error thrown. I think there is an argument which if you plot daily cost over time as history the daily total goes up and down and so it’s not total increasing but a changing daily total.

Reply:
As it's a daily value which resets each day, then it matches the definition for the `TOTAL_INCREASING`: https://developers.home-assistant.io/docs/core/entity/sensor/#available-state-classes

However, you're right if set to `TOTAL_INCREASING` it causes a warning: 
`2024-12-20 19:00:17.629 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.owl_intuition_cost_today (<class 'custom_components.owlintuition.sensor.OwlIntuitionSensor'>) is using state class 'total_increasing' which is impossible considering device class ('monetary') it is using; expected None or one of 'total'; Please update your configuration if your entity is manually configured, otherwise report it to the author of the 'owlintuition' custom integration`

Which suggests for `monetary` sensors `None` or `total` should work. The original code uses `None`. Let me check the original value of `None` again with the latest HA in case it was a bug with HA.

Reply:
`None` triggers an issue when looking at `Statistics`. Whereas `TOTAL` appears to work. I'd like to check how it behaves when it resets overnight.

Reply:
It also says in that statistic section…. In the developer notes

Entities representing a total amount

Entities tracking a total amount have a value that may optionally reset periodically, like this month's energy consumption, today's energy production, the weight of pellets used to heat the house over the last week or the yearly growth of a stock portfolio. The sensor's value when the first statistics is compiled is used as the initial zero-point.

I’m confused now!  It seems to contradict the Total_Increasing definition. 

I’ll leave it in your capable hands.

Reply:
Change appears to be good and doesn't trigger an odd behaviours when the sensor resets each day. @glpatcern